.circleci/scripts/validate-locales-only.ts

```diff
@@ -0,0 +1,48 @@
+const fs = require('fs').promises;
+const path = require('path');
+
+const BASE_PATH = path.resolve(__dirname, '..', '..');
+const CHANGED_FILES_PATH = path.join(
+  BASE_PATH,
+  'changed-files',
+  'changed-files.txt',
+);
+
+/**
+ * Reads the list of changed files from the git diff file.
+ *
+ * @returns {Promise<string[]>} An array of changed file paths.
```

Comment:
Types are not valid in TSDoc comments. The types are documented in TypeScript instead
```suggestion
 * @returns An array of changed file paths.
```

See here for more on TSDoc: https://tsdoc.org
This is what we use in TypeScript files rather than JSDoc, which is intended for JavaScript files.